pkg/config/resource.go

```diff
@@ -165,10 +165,26 @@ type ExternalName struct {
 	IdentifierFields []string
 }
 
+var ErrReferenceAlreadyExists = errors.New("reference for field already exists")
+
 // References represents reference resolver configurations for the fields of a
 // given resource. Key should be the field path of the field to be referenced.
 type References map[string]Reference
 
+// AddReference adds a reference configuration for the given field path.
+// The fieldPath is the Terraform field path of the field to be referenced.
+//
+// Example: "vpc_id" or "forwarding_rule.certificate_name" in case of nested
+// in another object.
+func (r References) AddReference(fieldPath string, ref Reference) error {
+	if _, ok := r[fieldPath]; ok {
+		return ErrReferenceAlreadyExists
```

Comment:
I don't love returning an error in this situation. My instinct is that the action of setting a reference should be idempotent; as you've implemented it all times it's invoked but the first will fail with an error. 

Maybe it's ok because it would happen at generation time, although I don't know if we could guarantee that.

Reply:
Hey @mbbush, what would be the resolution here? I am not sure if you requested some changes or not